src/hoc/RealtimeTemplates.tsx

```diff
@@ -0,0 +1,123 @@
+'use client'
+
+import { supabase } from '@/lib/supabase'
+import { selectTaskBoard } from '@/redux/features/taskBoardSlice'
+import { selectCreateTemplate, setTemplates } from '@/redux/features/templateSlice'
+import store from '@/redux/store'
+import { Token } from '@/types/common'
+import { ITemplate } from '@/types/interfaces'
+import { extractImgSrcs, replaceImgSrcs } from '@/utils/signedUrlReplacer'
+import { AssigneeType, TaskTemplate } from '@prisma/client'
+import { RealtimePostgresChangesPayload } from '@supabase/supabase-js'
+import { ReactNode, useEffect } from 'react'
+import { useSelector } from 'react-redux'
+
+interface RealTimeTemplateResponse extends ITemplate {
+  deletedAt: string
+}
+
+export const RealTimeTemplates = ({
+  children,
+  task,
+  tokenPayload,
+}: {
+  children: ReactNode
+  task?: ITemplate
+  tokenPayload: Token
+}) => {
+  const { templates } = useSelector(selectCreateTemplate)
+  const { token } = useSelector(selectTaskBoard)
+
+  const userId = tokenPayload?.internalUserId || tokenPayload?.clientId
+  const userRole = tokenPayload?.internalUserId
+    ? AssigneeType.internalUser
+    : tokenPayload?.clientId
+      ? AssigneeType.client
+      : undefined
+
+  if (!userId || !userRole) {
+    console.error(`Failed to authenticate a realtime connection for id ${userId} with role ${userRole}`)
+  }
+
+  const handleTemplatesRealTimeUpdates = (payload: RealtimePostgresChangesPayload<RealTimeTemplateResponse>) => {
+    if (payload.eventType === 'INSERT') {
+      let canUserAccessTask = payload.new.workspaceId === tokenPayload?.workspaceId
+      if (userRole === AssigneeType.client) {
+        canUserAccessTask = false
+      }
+      if (canUserAccessTask) {
+        templates
+          ? store.dispatch(
+              setTemplates([...templates, { ...payload.new, createdAt: new Date(payload.new.createdAt + 'Z') }]),
+            )
+          : store.dispatch(setTemplates([{ ...payload.new, createdAt: new Date(payload.new.createdAt + 'Z') }]))
+      }
+    }
+    if (payload.eventType === 'UPDATE') {
+      const updatedTemplate = payload.new
+      const isCreatedAtGMT = (updatedTemplate.createdAt as unknown as string).slice(-1).toLowerCase() === 'z'
+      if (!isCreatedAtGMT) {
+        // DB stores GMT timestamp without 'z', so need to append this manually
+        updatedTemplate.createdAt = ((updatedTemplate.createdAt as unknown as string) + 'Z') as unknown as Date
```

Comment:
Add support for updatedAt, since we sort by most recently modified first

Reply:
I think timestamps in templates table are stored differently than tasks table, *with* timezone info.

Could you please confirm this

Reply:
* I confirmed this, this is fixed in a recent migration to keep things consistent with tasks table